Convert ResetCustomerPasswordFailedTest to MFTF

rogyar opened the pull request:
### Description (*)
This PR introduces MFTF test that covers the following scenario:
- Guest user goes to the forgot password page, submits an email address
- User sees the success message
- User goes to the forgot password page once again and submits an email address
- The system shows error about too many requests for password resets

### Fixed Issues (if relevant)
1. #379 : Convert ResetCustomerPasswordFailedTest to MFTF

okolesnyk commented on app/code/Magento/Customer/Test/Mftf/Test/StorefrontResetCustomerPasswordFailedTest.xml:
```diff
@@ -0,0 +1,47 @@
+        <click stepKey="clickResetPassword1" selector="{{StorefrontForgotPasswordSection.resetMyPasswordButton}}"/>
+        <seeInCurrentUrl stepKey="seeInSignInPage" url="account/login"/>
+        <waitForElementVisible selector="{{StorefrontCustomerLoginMessagesSection.successMessage}}" stepKey="waitForSuccessMessage" />
+        <see stepKey="seeSuccessMessage" userInput="If there is an account associated with $$customer.email$$ you will receive an email with a link to reset your password." selector="{{StorefrontCustomerLoginMessagesSection.successMessage}}"/>
```
Please create an actionGroup for this. Adding you patch with example how we should build our actionGroups.
[suggest.patch.zip](https://github.com/magento/magento-functional-tests-migration/files/3015308/suggest.patch.zip)

okolesnyk commented on app/code/Magento/Customer/Test/Mftf/Test/StorefrontResetCustomerPasswordFailedTest.xml:
```diff
@@ -0,0 +1,47 @@
+        <seeInCurrentUrl stepKey="seeInForgotPasswordPage" url="account/forgotpassword"/>
+        <waitForElementVisible selector="{{StorefrontCustomerLoginMessagesSection.errorMessage}}" stepKey="waitForFailMessage" />
+        <see stepKey="seeFailMessage" userInput="We received too many requests for password resets. Please wait and try again later or contact hello@example.com." selector="{{StorefrontCustomerLoginMessagesSection.errorMessage}}"/>
+    </test>
```
After you apply created actionGroup your test should look like this:
```
        <actionGroup ref="StorefrontCustomerResetPasswordActionGroup" stepKey="resetPasswordFirstAttempt">
            <argument name="email" value="$$customer.email$$" />
            <argument name="message" value="If there is an account associated with $$customer.email$$ you will receive an email with a link to reset your password." />
        </actionGroup>
        <actionGroup ref="StorefrontCustomerResetPasswordActionGroup" stepKey="resetPasswordSecondAttempt">
            <argument name="email" value="$$customer.email$$" />
            <argument name="message" value="We received too many requests for password resets. Please wait and try again later or contact hello@example.com." />
            <argument name="messageType" value="error" />
        </actionGroup>
```

rogyar commented:
Hi @okolesnyk. Thanks a lot for your recommendations. I've updated the code according to suggestions.
There was one minor problem: the success message is shown on the customer login page but the error message is shown on the forgot password page. So, I just removed URL assertion from the action group.

okolesnyk commented:
Hi @rogyar 
Can please take a look at my last comment I left https://github.com/magento/magento-functional-tests-migration/pull/682#pullrequestreview-219711371

To not remove URL verification I wanted to split Action Group You created to 2 smaller.
And send URL as an argument to an action group.

rogyar commented:
@okolesnyk got it. I've added the additional action group. Thank you
